org-roam.el

```diff
@@ -134,6 +134,27 @@ space-delimited strings.
            (symbol)))
   :group 'org-roam)
 
+(defcustom org-roam-tag-sources '(prop)
+  "Sources to obtain tags from.
+
+It should be a list of symbols representing any of the following
+extraction methods:
+
+1. 'prop: This extracts tags from the =#+ROAM_TAGS=
+   property.  Tags are space delimited, and can be multi-word
+   using double quotes.
+
+2. 'all-directories: All sub-directories relative to
+   =org-roam-directory= are extracted as tags.  That is, if a file
+   is located at relative path =foo/bar/file.org=, the file will
+   have tags =foo= and =bar=.
+
+3. 'last-directory: Extracts the last directory relative to
+   =org-roam-directory= as the tag.  That is, if a file is located
+   at relative path =foo/bar/file.org=, the file will have tag
+   =bar=."
```

Comment:
```suggestion
  `prop'
    Extract tags from the #+ROAM_TAGS property.
    Tags are space delimited.
    Tags may contain spaces if they are double-quoted.
    e.g. #+ROAM_TAGS: tag \"tag with spaces\"

  `all-directories'
    Extract sub-directories relative to `org-roam-directory'.
    That is, if a file is located at relative path foo/bar/file.org,
    the file will have tags \"foo\" and \"bar\".

  `last-directory'
    Extract the last directory relative to `org-roam-directory'.
    That is, if a file is located at relative path foo/bar/file.org,
    the file will have tag \"bar\"."
```

Elisp convention is to quote symbols in docstrings.

```

Reply:
I'd also consider changing `all-directories` to `sub-directories` and `last-directory` to `parent-directory`.